config/routes.rb

```diff
@@ -18,7 +18,14 @@
   resources :homeworks
 
   resources :invitations, only: [ :new, :create, :edit, :update ]
-  resources :tasks, only: [ :show, :update ]
+  resources :tasks, only: [ :show, :update ] do
+    member do
+      get 'switch_chat'
+    end
+  end
```

Comment:
Can it be rewritten using Ruby blocks?
[This article](http://awaxman11.github.io/blog/2013/08/05/what-is-the-difference-between-a-block/) may be helpful.